ale_linters/scala/scalac.vim

```diff
@@ -0,0 +1,61 @@
+" vim: set et:
+" Author: Zoltan Kalmar - https://github.com/kalmiz
+" Description: Basic scala support using scalac
+
+if exists('g:loaded_ale_linters_scala_scalac')
+    finish
+endif
+
+let g:loaded_ale_linters_scala_scalac = 1
+
+function! ale_linters#scala#scalac#Handle(buffer, lines)
+    " Matches patterns line the following:
+    "
+    " /var/folders/5q/20rgxx3x1s34g3m14n5bq0x80000gn/T/vv6pSsy/0:26: error: expected class or object definition
+    let pattern = '^.\+:\(\d\+\): \(\w\+\): \(.\+\)'
+    let output = []
+    let ln = 0
+
+    for line in a:lines
+        let ln = ln + 1
+        let l:match = matchlist(line, pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let text = l:match[3]
+        let type = 'W'
+        if l:match[2] == 'error'
+            let type = 'E'
+        endif
+        let col = 0
+        if exists('a:lines[ln+1]')
```

Comment:
Can you not check `if ln + 1 < len(a:lines)` instead? It looks like you're doing a bounds check. If the string is empty for some reason, then `stridx` below will just return `-1` anyway.